### PR TITLE
JDK-8157682: @inheritDoc doesn't work with @exception

### PR DESCRIPTION
**Pull request by jonathan-gibbons**
Please review a fix to improve the handling of the legacy `@exception` tag.

A patch for this was [previously submitted](https://mail.openjdk.java.net/pipermail/javadoc-dev/2019-September/001139.html). However, that patch introduced an explicit new `ExceptionTaglet` class, whereas the fix here just reuses the existing `ThrowsTaglet`, registering it under an additional name.

The test in the original patch is enhanced to check all 4 combinations of (throws, exception) in the base class and (throws, exception) in the subclass.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8157682](https://bugs.openjdk.java.net/browse/JDK-8157682): @inheritDoc doesn't work with @exception

### Reviewers
 * [Pavel Rappo](https://openjdk.java.net/census#prappo) (@pavelrappo - **Reviewer**)

### Contributors
 * Yano, Masanori `<yano-masanori@jp.fujitsu.com>`
 * Jonathan Gibbons `<jjg@openjdk.org>`

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/2818/head:pull/2818`
`$ git checkout pull/2818`

### CODE REVIEW

**Review by pavelrappo — CHANGES_REQUESTED** [review]
This commit changes the order of block tags in the output. The fact that no tests failed when I ran them surprised me. I think we should both fix that order and introduce a test for it.

**Comment by jonathan-gibbons** [comment]
/contributor add Yano, Masanori <yano-masanori@jp.fujitsu.com>

File: src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/taglets/TagletManager.java [hunk 1]
```diff
@@ -584,7 +584,10 @@ private void initTaglets() {
 
             if (t.isBlockTag()) {
                 for (Location l : t.getAllowedLocations()) {
-                    blockTagletsByLocation.get(l).add(t);
+                    List<Taglet> list = blockTagletsByLocation.get(l);
+                    if (!list.contains(t)) {
+                        list.add(t);
+                    }
```
[the remark this answers is not in the document]
Comment (jonathan-gibbons) [reply src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/taglets/TagletManager.java:590]: yeah, but ... this is just one-off init ... it's the "standard" problem of list vs. sets, and list-y sets vs. set-y lists.

**Comment by jonathan-gibbons** [comment]
> This commit changes the order of block tags in the output. The fact that no tests failed when I ran them surprised me. I think we should both fix that order and introduce a test for it.

Can you give more details?  Did you observe that anywhere, or are you inferring that from the code somewhere?

**Comment by pavelrappo** [comment]
> > This commit changes the order of block tags in the output. The fact that no tests failed when I ran them surprised me. I think we should both fix that order and introduce a test for it.
> 
> Can you give more details? Did you observe that anywhere, or are you inferring that from the code somewhere?

I saw that difference in the produced documentation for JDK. I suggest you build documentation for "before" and "after", each time with the `-notimestamp` option, and compare them; you should see what I saw.

**Comment by jonathan-gibbons** [comment]
> > > This commit changes the order of block tags in the output. The fact that no tests failed when I ran them surprised me. I think we should both fix that order and introduce a test for it.
> > 
> > 
> > Can you give more details? Did you observe that anywhere, or are you inferring that from the code somewhere?
> 
> I saw that difference in the produced documentation for JDK. I suggest you build documentation for "before" and "after", each time with the `-notimestamp` option, and compare them; you should see what I saw.

Wow. Good catch! That was unexpected! Will investigate!

**Review by pavelrappo — APPROVED** [review]
I can see that you propose to add a test for the output order of block tags separately: https://git.openjdk.java.net/jdk/pull/2835 (JDK-8263043).

Looks good.

**Pull request closed without merging**